src/main/java/ml/shifu/shifu/core/binning/CategoricalBinning.java

```diff
@@ -205,8 +205,8 @@ public void stringToObj(String objValStr) {
     public String objToString() {
         try {
             return super.objToString() + Character.toString(FIELD_SEPARATOR) + Boolean.toString(isValid)
-                    + Character.toString(FIELD_SEPARATOR) + StringUtils.join(categoricalVals, SETLIST_SEPARATOR)
-                    + Character.toString(FIELD_SEPARATOR) + Base64Utils.base64EncodeFromBytes(this.hyper.getBytes());
+                    + Character.toString(FIELD_SEPARATOR) + Base64Utils.base64EncodeFromBytes(this.hyper.getBytes())
+                    + Character.toString(FIELD_SEPARATOR) + StringUtils.join(categoricalVals, SETLIST_SEPARATOR);
```

Comment:
The value of `categoricalVals` may has special charator `FIELD_SEPARATOR`. We put this field to the end of string value.